File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/dialogue/dto/SttDialogueResponse.java
package com.study.webflux.rag.application.dialogue.dto;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "음성 입력 기반 대화 결과")
public record SttDialogueResponse(
	@Schema(description = "음성에서 추출한 텍스트", example = "오늘 회의 일정 알려줘")
	String transcription,

	@Schema(description = "전사 텍스트를 기반으로 생성한 대화 답변", example = "오늘 회의는 오후 2시에 시작합니다.")
	String response
) {
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/inbound/web/dialogue/dto/SttDialogueResponse.kt
package com.miyou.app.infrastructure.inbound.web.dialogue.dto

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "STT 입력 결과 응답")
data class SttDialogueResponse(
    @field:Schema(description = "요청한 오디오에서 추출된 문자", example = "안녕하세요, 어떻게 도와드릴까요?")
    val transcription: String,
    @field:Schema(description = "STT 전송 후 생성된 응답", example = "안녕하세요! 무엇을 도와드릴까요?")
    val response: String,
)
